### DISCUSSION TITLE
More than one shutdownCommand

### DISCUSSION THREAD

**Discussion by tanmaster** [AlexGustafsson/homebridge-wol#167] [Questions] [answered]
I have a very edgy case in which I run dual boot windows and linux, both needing seperate shutdown commands. I tried to chain both commands together using a ';' or a '||' and these methods work well when the commandchain is issued over a normal shell. It is certain that one command will always fail and the other will succeed when its working correctly.
However in homebridge-wol, the library that is used for issuing the command (the execSync function) immediately returns once the first part of the command chain fails. This means that I can't chain these two commands together :-(

Is there any solution for this problem? I could not think of anything without having to modify the code, but I thought I should ask before I make a pull request...

**Comment by AlexGustafsson** [comment, marked as answer]
It sounds weird why `execSync` would behave that way as it should invoke a full shell and then rely on it for handling things correctly. Could you try to use a shutdown command like `/bin/sh -c 'my-failing-command || my-working-command'`? That way there really should not be anything different from your own terminal or what the plugin is doing.

**Reply by tanmaster** [reply]
Good thing I asked, that did indeed work for me :-)

Thanks 🙏
